src/raster/buffer.rs

```diff
@@ -0,0 +1,86 @@
+use crate::raster::GdalType;
+
+#[cfg(feature = "ndarray")]
+use ndarray::Array2;
+
+/// A 2-D array backed by it's `size` (cols, rows) and a row-major `Vec<T>` and it's dimensions.
+#[derive(Debug, Clone, PartialEq)]
+pub struct Buffer<T> {
+    pub size: (usize, usize),
+    pub data: Vec<T>,
+}
+
+impl<T: GdalType> Buffer<T> {
+    /// Construct a new buffer from `size` (`(cols, rows)`) and `Vec<T>`.
+    ///
+    /// # Notes
+    /// Assumes `size.0 * size.1 == data.len()`.
+    pub fn new(size: (usize, usize), data: Vec<T>) -> Buffer<T> {
+        debug_assert!(
```

Comment:
```suggestion
        assert!(
```

I don't think anything good can come out of passing in the wrong shape.